File: Chapter-6-Constructor-and-Destructor-Functions/001-a-sample-program-to-introduce-parameterized-constructor.cpp
// a sample program to introduce parameterized constructor
#include <bits/stdc++.h>

using namespace std;

class integerNumber
{
    int m, n;

public:
    // integerNumber(int, int); // Parameterized constructor declaration
    integerNumber(int x, int y)
    {
        m = x;
        n = y;
    }
    void displayIntegerNumber() // Member function to display m and n
    {
        cout << "m = " << m << endl;
        cout << "n = " << n << endl;
    }
};

// integerNumber::integerNumber(int x, int y)
// {
//     m = x;
//     n = y;
// }

int main()
{
    integerNumber A(0, 100);
    A.displayIntegerNumber();
}

// output
// m = 0
// n = 100

File: Chapter-6-Constructor-and-Destructor-Functions/001-a-sample-program-to-introduce-parameterized-constructor.py
class IntegerNumber:
    def __init__(self, x, y):
        self.m = x
        self.n = y
    
    def display_integer_number(self):
        print(f'm = {self.m} and n = {self.n}')
    
# Create an object of the class
obj = IntegerNumber(10, 20)
obj.display_integer_number()